react/components/TrackEventsWrapper.tsx

```diff
@@ -0,0 +1,54 @@
+import { forEach } from 'ramda'
+import { PureComponent } from 'react'
+import ReactDOM from 'react-dom'
+
+interface Props {
+  events?: string[]
+  id?: string
+}
+
+const sendEvent = (id: string, event: string) => {
+  window.vtex.NavigationCapture.sendEvent(
+    'pageComponentInteraction',
+    {
+      pageComponentInteraction : {
+        componentId: id,
+        interactionType: event,
+      }
+    }
+  )
+}
+
+class TrackEventsWrapper extends PureComponent<Props> {
+
+  public componentDidMount(){
+    const { id, events } = this.props
+    if(events && events.length && id){
+      const element = ReactDOM.findDOMNode(this)
+      if(element && element.addEventListener){
+        forEach(
+          event => element.addEventListener(event, () => sendEvent(id, event))
+        , events)
+      }
+    }
+  }
+
+  public componentWillUnmount(){
+    const { id, events } = this.props
+    if(events && events.length && id){
+      const element = ReactDOM.findDOMNode(this)
+      if(element && element.addEventListener){
+        forEach(
+          event => element.removeEventListener(event, () => sendEvent(id, event))
```

Comment:
this won't work, because you are creating a new function every step of the loop, and these won't be the same reference as the one you registered with `addEventListener`.

instead you should create a property on this class with the event handler and register and unregister passing in the same function. you could use [this class](https://github.com/vtex/pixel-manager/blob/951350bd1365c52903c9d4aea0463628a344d331/react/PixelIFrame.tsx#L21-L49) as a reference